src/otx/core/data/pre_filtering.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Pre filtering data for OTX."""
+
+from __future__ import annotations
+
+import warnings
+from typing import TYPE_CHECKING
+
+from datumaro.components.annotation import Annotation, Bbox, Polygon
+from datumaro.components.dataset import Dataset as DmDataset
+
+if TYPE_CHECKING:
+    from datumaro.components.dataset_base import DatasetItem
+
+
+def pre_filtering(dataset: DmDataset, data_format: str) -> DmDataset:
+    """Filtering invalid data in datumaro dataset."""
+    dataset = DmDataset.filter(dataset, is_non_empty_item)
+    dataset = DmDataset.filter(dataset, is_valid_annot, filter_annotations=True)
+
+    return remove_unused_labels(dataset, data_format)
+
+
+def is_non_empty_item(item: DatasetItem) -> bool:
+    """Return whether DatasetItem's annotation is non-empty."""
+    if not (item.subset == "train" and len(item.annotations) == 0):
+        return True
+    msg = "There are empty annotation items in train set, they will be filtered out before training."
+    warnings.warn(msg, stacklevel=2)
+    return False
+
+
+def is_valid_annot(item: DatasetItem, annotation: Annotation) -> bool:
+    """Return whether DatasetItem's annotation is valid."""
+    del item
```

Comment:
Why do you `del item`?

Reply:
In this function, item is not used, but required. If we delete line 37 then, ruff will complaining about unused argument.

Reply:
Please just add ruff noqa

Reply:
Done by [e36ab20](https://github.com/openvinotoolkit/training_extensions/pull/2823/commits/e36ab20751d26fdb948a5884afb30c3c1f17c7d9)